### PR TITLE
Mobile 3903

### PR DESCRIPTION
**Pull request by dpalou**

### CODE REVIEW

File: src/core/features/courses/services/handlers/course-link.ts [hunk 1]
```diff
@@ -167,95 +167,108 @@ export class CoreCoursesCourseLinkHandlerService extends CoreContentLinksHandler
...
+        }
+    }
+
+    /**
+     * Check if user can access the course.
+     *
+     * @param courseId Course ID.
+     * @return Promise resolved with boolean: whether user can access the course.
+     */
+    protected async canAccess(courseId: number): Promise<boolean> {
```
Comment (crazyserver) [inline src/core/features/courses/services/handlers/course-link.ts:239]: There's a method called promiseWorks that returns true or false on catch, you're free to use it ;-)

File: src/core/features/courses/services/handlers/course-link.ts [hunk 2]
```diff
@@ -319,21 +329,18 @@ export class CoreCoursesCourseLinkHandlerService extends CoreContentLinksHandler
         // Check if user is enrolled in the course.
         await CoreUtils.ignoreErrors(CoreCourses.invalidateUserCourses());
         try {
-            return CoreCourses.getUserCourse(courseId);
-        } catch {
+            return await CoreCourses.getUserCourse(courseId);
+        } catch (error) {
```
Comment (crazyserver) [inline src/core/features/courses/services/handlers/course-link.ts:333]: Are you using this error?

**Review by crazyserver — CHANGES_REQUESTED** [review]
Thanks Dani! Two minor things found, feel free to merge when done.

File: src/core/features/courses/services/handlers/course-link.ts (hunk 1 above)
Comment (dpalou) [reply src/core/features/courses/services/handlers/course-link.ts]: Right, nice catch.

File: src/core/features/courses/services/handlers/course-link.ts (hunk 2 above)
Comment (dpalou) [reply src/core/features/courses/services/handlers/course-link.ts]: Nope, I added this to debug and then I forgot to remove it.

**Review by crazyserver — APPROVED** [review]

**Pull request merged**